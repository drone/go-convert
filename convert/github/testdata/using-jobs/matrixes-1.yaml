# https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#example-expanding-configurations

jobs:
  example_matrix:
    strategy:
      matrix:
        os: [windows-latest, ubuntu-latest]
        node: [12, 14, 16]
        include:
          - os: windows-latest
            node: 16
            npm: 6
    runs-on: ${{ matrix.os }}
    steps:
      - uses: actions/setup-node@v3
        with:
          node-version: ${{ matrix.node }}
      - if: ${{ matrix.npm }}
        run: npm install -g npm@${{ matrix.npm }}
      - run: npm --version
