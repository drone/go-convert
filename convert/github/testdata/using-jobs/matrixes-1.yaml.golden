kind: pipeline
spec:
  stages:
  - name: example_matrix
    spec:
      platform:
        arch: amd64
        os: linux
      runtime:
        spec: {}
        type: cloud
      steps:
      - spec:
          uses: actions/setup-node@v3
          with:
            node-version: ${{ matrix.node }}
        type: action
      - spec:
          run: npm install -g npm@${{ matrix.npm }}
        type: script
      - spec:
          run: npm --version
        type: script
    strategy:
      spec:
        axis:
          node:
          - "12"
          - "14"
          - "16"
          os:
          - windows-latest
          - ubuntu-latest
        include:
        - node: "16"
          npm: "6"
          os: windows-latest
      type: matrix
    type: ci
version: 1
